Steer/src/MCKinematicsReader.cxx

```diff
@@ -0,0 +1,53 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "Steer/MCKinematicsReader.h"
+#include <TChain.h>
+#include <vector>
+#include "FairLogger.h"
+
+using namespace o2::steer;
+
+bool MCKinematicsReader::init(std::string const& name)
+{
+  auto context = DigitizationContext::loadFromFile(name.c_str());
+  if (!context) {
+    return false;
+  }
+  mInitialized = true;
+  mDigitizationContext = context;
+
+  // get the chains to read
+  mDigitizationContext->initSimKinematicsChains(mInputChains);
+
+  // load the kinematics information
+  mTracks.resize(mInputChains.size());
+
+  // load the kinematics information
+  for (int source = 0; source < mInputChains.size(); ++source) {
```

Comment:
this may require lot of memory, even if the user don't need full MC info but e.g. signal only. Perhaps we should use lazy init, where the particular source is loaded automatically on 1st query?

Reply:
Yes. Good idea.